### PR TITLE
git: compile diff-highlight

### LINKED ISSUE #15588: git 2.13.3 and up: missing diff-highlight
After updating `git` `diff-highlight` is no longer available.

`git 2.13.2`:
```
$ ls /usr/local/opt/git/share/git-core/contrib/diff-highlight/
t  Makefile  README  diff-highlight
```

`git 2.13.3` and `git HEAD`:
```
$ ls /usr/local/opt/git/share/git-core/contrib/diff-highlight/
t  DiffHighlight.pm  Makefile  README  diff-highlight.perl
```

### PR DESCRIPTION
**Pull request by ghostbar**
This would fix #15588 correctly.

### CODE REVIEW

**Comment by ilovezfs** [comment]
As mentioned on https://github.com/Homebrew/homebrew-core/issues/15588#issuecomment-315140167 I don't really think it makes sense to be compiling a file in `contrib` forevermore.

**Comment by ilovezfs** [comment]
Ah, I see we're already doing that for a subset of the others, so what's one more? If we are going to compile it, though, shouldn't we also install it in `bin`?

**Comment by ghostbar** [comment]
I don't know. It was already being compiled. I'm just trying to have the same functionality it had before.

**Comment by ilovezfs** [comment]
@ghostbar how do you actually use it? Are you specifying a full path to it in you git config or something?

**Comment by ghostbar** [comment]
@ilovezfs https://github.com/ghostbar/dotfiles/blob/master/zshfiles/zprofile#L26

And that's how it's mostly recommended to be used on the websites that make a reference to `diff-highlight`.

**Comment by ilovezfs** [comment]
@ghostbar yikes. Yeah we should just install it in `bin` in that case.

File: Formula/git.rb [hunk 1]
```diff
@@ -113,6 +113,11 @@ def install
       bin.install "git-credential-osxkeychain"
       system "make", "clean"
     end
+    
+    # Install diff-highlight
+    cd "contrib/diff-highlight" do
+      system "make"
```
Comment (ilovezfs) [inline Formula/git.rb]:
```
      bin.install "diff-highlight"
```

**Comment by ghostbar** [comment]
So that would made it, right? @ilovezfs

**Comment by ilovezfs** [comment]
@ghostbar so you can test that by installing, and just deleting`$(brew --prefix git)/share/git-core/contrib/diff-highlight` from your profile.

**Comment by jasonkarns** [comment]
Installing `diff-highlight` in bin would automatically put it in `PATH`, right? I don't think that's desirable. The understanding for `contrib/` utilities is that they are _available_ but aren't automatically part of a system installation.

Whereas simply doing the compilation as part of install makes sense, because it doesn't force that utility into anyone's `PATH`, and just simplifies the ability to opt-in.

For example, the `new-workdir` contrib utility, the instructions are to place the `contrib/workdir` directory in `PATH`. I wouldn't want _all_ of the contrib utilities added to `PATH`, and neither would it make sense for some of them to be added, but not others. So my vote is to compile the utilities such that opt-in usage is trivial. But not to place it in `bin`.

**Comment by ilovezfs** [comment]
@jasonkarns did you actually look at the handling of other contrib items in the current formula?

**Comment by jasonkarns** [comment]
Nope! Because again, I would not have assumed that homebrew violate the longstanding convention of how contrib utils are handled. Current behavior seems sporadic. Some utilities are installed by default (subtree, credential helpers), others are opt-in (completions, https), and still others are ignored (workdir, diff-highlight). I wouldn't say that's the most intuitive installation of contrib utilities.

**Comment by ilovezfs** [comment]
@jasonkarns yeah, I agree, but I don't think we need to upend the apple cart, so to speak, over this one file.

**Comment by jasonkarns** [comment]
@ilovezfs acknowledged. It seems to be working. Just puts homebrew in the unenviable position of deciding for its users what it deems default vs opt-in vs manual.

I could see a case being made for the OSX-specific utilities (keychain credential helper) and "80% desirable" completions utility.

But to continue arguing against bin for diff-highlight: this utility would be a brand new _addition_ to homebrew's git installation. So its merit as a default utility (or opt-in using a --with- flag) would need to be determined, yes? Whereas simply compiling it would maintain the status quo. (That is, the utility is functional after installation, but needs invoked directly or with manual `PATH` manipulation.)

**Comment by ilovezfs** [comment]
@jasonkarns I think we could literally talk ourselves into or out of any given subset. I guess I'm fine with just running `make`.

**Comment by ghostbar** [comment]
I'm fine with just `make` as well since it was the previous behaviour.

**Comment by ilovezfs** [comment]
@ghostbar Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**